zbark/debark_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package zbark
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/uber-go/zap"
+
+	"github.com/uber-common/bark"
+)
+
+func newDebark() (zap.Logger, *bytes.Buffer) {
+	buf := &bytes.Buffer{}
+	logger := logrus.New()
+	logger.Out = buf
+	logger.Level = logrus.DebugLevel
+	return Debarkify(bark.NewLoggerFromLogrus(logger)), buf
+}
+
+func TestDebark_CastNoop(t *testing.T) {
+	orig := zap.NewJSON(zap.DebugLevel)
+	assert.Equal(t, orig, Debarkify(Barkify(orig)))
```

Comment:
This will do a deep equals, I think you want to do a pointer check here, `assert.True(t, orig == Debarkify(...), ...)`